src/lib/mac-address.ts

```diff
@@ -0,0 +1,33 @@
+import * as fs from 'mz/fs';
+import * as path from 'path';
+
+import log from './supervisor-console';
+
+export class MacAddressError extends Error {}
+
+export async function getAll(sysClassNet: string): Promise<string | undefined> {
+	try {
+		// read the directories in the sysfs...
+		const interfaces = await fs.readdir(sysClassNet);
+
+		// read the directories in the sysfs...
+		return (
+			await Promise.all(
+				interfaces.map(async (intf) => {
+					try {
+						const file = path.join(sysClassNet, intf, 'address');
+						const addr = await fs.readFile(file, 'utf8');
+						return addr.split('\n')[0].trim().toUpperCase();
+					} catch {
+						return '';
```

Comment:
Is this expected to happen a lot, or is it an exceptional circumstance? Basically I'm wondering if it points to other issues on-device, and we should report it in the logs.

Reply:
Yeah I think in that case we want to add a log as breadcrumbs for why this failed.